clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -2569,6 +2577,23 @@ public long getRegionHybridOffsetLag(int regionId) {
     return StatsErrorCode.ACTIVE_ACTIVE_NOT_ENABLED.code;
   }
 
+  /**
+   * Once leader is marked completed, immediately send a heart beat message to the local VT such that
+   * followers don't have to wait till the periodic heartbeat to know that the leader is completed
+   *
+   * Also update veniceWriter of this status, such that the SOS messages sent from the leader to the
+   * followers can have this status.
+   */
+  @Override
+  protected void sendInstantHeartBeatAndUpdateVeniceWriter(
+      PartitionConsumptionState partitionConsumptionState,
+      PubSubTopicPartition pubSubTopicPartition) {
+    boolean isLeaderCompleted = partitionConsumptionState.isCompletionReported();
+    veniceWriter.get()
+        .sendHeartbeat(pubSubTopicPartition, null, DEFAULT_LEADER_METADATA_WRAPPER, true, 0, isLeaderCompleted);
+    veniceWriter.get().setPartitionLeaderCompletionStatus(pubSubTopicPartition.getPartitionNumber(), isLeaderCompleted);
```

Comment:
It looks weird to me to pass `isLeaderCompleted` into `sendHeartbeat` and also into `setPartitionLeaderCompletionStatus`... I think we should either:

1. Say that the `VW` needs to remember the completion state, in which case we should call `setPartitionLeaderCompletionStatus(partition, isLeaderCompleted )` first, and then `sendHeartbeat` (but without passing `isLeaderCompleted`)... OR
2. Say that the `LFSIT` is responsible for remembering, in which case it should always pass the param and `VW` should not carry that state at all (and thus we delete the `setPartitionLeaderCompletionStatus` function and the data structure inside `VW`)...

Reply:
Great point. Changed it. Chose to set the state in `VW` and then call `sendHeartbeat` without passing `isLeaderCompleted` as VW would need the state to send `SOS` as well.